Packs/DeCYFIR/Integrations/DecyfirEventCollector/DecyfirEventCollector.py

```diff
@@ -0,0 +1,398 @@
+    """
+    Determine next fetch time based on the latest event timestamp.
+
+    Args:
+        events (List[Dict[str, Any]]): List of fetched events.
+        previous_time (Optional[datetime]): Last recorded fetch time.
+        event_type (str): Type of event.
+
+    Returns:
+        Optional[str]: ISO formatted datetime for the next fetch cycle.
+    """
+    # Extract valid datetimes only (filter out None explicitly)
+    times: List[datetime] = [t for e in events if (t := extract_event_time(e, event_type)) is not None]
+
+    if not times:
+        # Fall back to previous time if no valid timestamps found
+        return previous_time.isoformat() if previous_time else None
+
+    # mypy-safe: times only contains datetime objects
+    latest: datetime = max(times)
+    next_time: datetime = latest + timedelta(milliseconds=1)
+    return next_time.isoformat()
+
+
+""" FETCH EVENTS """
+
+
+def fetch_events(
+    client: Client,
+    last_run: dict[str, Any],
+    first_fetch_time: datetime,
+    event_types_to_fetch: list[str],
+    max_events_per_fetch: dict[str, int],
+) -> tuple[dict[str, Any], list[dict[str, Any]]]:
+    """
+    Fetch and process events for all configured event types.
+
+    Args:
+        client (Client): API client instance.
+        last_run (Dict[str, Any]): State of last execution.
+        first_fetch_time (datetime): Initial fetch baseline.
+        event_types_to_fetch (List[str]): List of log types to fetch.
+        max_events_per_fetch (Dict[str, int]): Max number of events per type.
+
+    Returns:
+        Tuple[Dict[str, Any], List[Dict[str, Any]]]: Updated run state and fetched events.
+    """
+    demisto.debug("Starting fetch_events")
+    current_run: dict[str, dict[str, Any]] = {}
+    all_events: list[dict[str, Any]] = []
+
+    for event_type in event_types_to_fetch:
+        demisto.debug(f"Fetching for {event_type}")
+        after = get_after_param(last_run, event_type, first_fetch_time)
+        events = client.search_events(event_type, max_events_per_fetch.get(event_type, MAX_EVENTS_PER_FETCH), after)
+
+        if events:
+            if event_type == ACCESS_LOGS:
+                update_fetched_event_ids(current_run, event_type, events)
+                events = remove_duplicate_logs(events, last_run, event_type)
+
+            add_event_fields(events, event_type)
+            all_events.extend(events)
+
+        latest_time = compute_next_fetch_time(events, first_fetch_time, event_type)
+        current_run.setdefault(event_type, {})["next_fetch_time"] = latest_time
+
+    demisto.debug(f"Fetch complete. Total events: {len(all_events)}")
+    return current_run, all_events
+
+
+""" COMMANDS """
+
+
+def test_module(client: Client) -> str:
+    """
+    Test connectivity and authentication.
+
+    Returns:
+        str: "ok" if successful.
+    """
+    fetch_events(client, {}, datetime.now(tz=UTC) - timedelta(minutes=5), [ACCESS_LOGS], {ACCESS_LOGS: 1})
+    return "ok"
```

Comment:
```suggestion
                  log_event.get("uid") not in last_run.get(event_type, {}).get("fetched_events_ids", []])]
```

This will fail if for some reason fetched_events_ids is not in the last_run